intelliflex_UI/src/services/users.service.ts

```diff
@@ -0,0 +1,105 @@
+import { Injectable } from '@angular/core';
+import { Response } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
+import { Headers, RequestOptions } from '@angular/http';
+import { HttpInterceptorService, RESTService } from '@covalent/http';
+import { MOCK_API } from '../config/api.config';
+import { Prod } from '../app/dashboard/dashboard.model';
+export interface IUser {
+  displayName: string;
+  id: string;
+  email: string;
+  created: Date;
+  lastAccess: Date;
+  siteAdmin: number;
+  
+}
+
+@Injectable()
+export class UsersService  {
+    public baseUrl: string;
+    public handleError: any;
+    public extractData: any;
+    public delete: any;
+    public query: any;
+    public get: any;;
+    public create: any;
+    public update: any;
+    public options: RequestOptions;
+  constructor(private _http: HttpInterceptorService) {
+    this.handleError = this.getHandleErrorFunction();
+    this.baseUrl = MOCK_API;
+  this.extractData = this.getExtractDataFunction();
+  }
+ public getHandleErrorFunction() {
```

Comment:
Can we have a generic error handler function to which we can send an initialise error object?
If you can come up with such a function we can start using it in all UI code.
Also we should externalise the error message that we are setting for statusText variable.